packages/server-wallet/src/wallet/multi-threaded-wallet/manager.ts

```diff
@@ -67,17 +84,23 @@ export class WorkerManager {
 
     const worker = await this.pool.acquire().promise;
 
-    const resultPromise = new Promise<T>((resolve, reject) =>
-      worker.once('message', (response: Either<Error, T>) => {
+    const resultPromise = new Promise<T>((resolve, reject) => {
+      const handler = (response: Either<Error, T>) => {
         this.pool?.release(worker);
 
         if (isLeft(response)) {
           reject(response.left);
-        } else {
+        } else if (isRight(response)) {
+          // WARNING: The correctness of this implementation relies on the fact that a worker
+          // only handles one concurrent operation.
+          // If a worker handles concurrent operations, then we should wait until we get a reply
+          // with the same message id.
+          worker.off('message', handler);
```

Comment:
I am trying to map the change here to 

> listen to 'message' events until the event contains the result of the API call

from the PR description. Is that the right mapping? If so, the only check is the `isRight()` check.

If left and right are mutually exclusive, I think this change wouldn't add any new functionality.

I think I would understand this,  if the response is being considered as left (error) , right (response), and neither left nor right (which you are using for relaying events). Am I in the right ballpark?

Reply:
> I think I would understand this, if the response is being considered as left (error) , right (response), and neither left nor right (which you are using for relaying events). Am I in the right ballpark?

Yes, that's correct. Perhaps the types could be "cleaned up", but the alternative I thought of was messy:
```
type WorkerWalletEvent = {
  type: 'WalletEvent';
  value: WalletEvent;
}

type APIResult = {
  type: 'APIResult';
  value: Either<Error, SingleOrMultipleChannelResult>
}

type WorkerMessage = WorkerWalletEvent | WorkerAPIResult
```

There is now an onion-like type.

Maybe an appealing alternative is
```
// snip
type WorkerMessage = WorkerWalletEvent | APIError | APIResult
```

which is, funny enough, the type in the codebase, if we have
```
type APIError = Left<Error>
type APIResult = Right<SingleOrMultipleChannelResult>
```